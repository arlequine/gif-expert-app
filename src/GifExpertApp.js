import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import PropTypes from 'prop-types'
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
  // const categories = ['One punch', 'samurai X', 'Dragon ball', 'One piece'];
  const [categories, setCategories] = useState(['One piece']);

  // const handleAddCategory = () => {
  //   const newCategories = [...categories, 'HunterxHunter'];
  //   setCategories(newCategories);
  // };

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      {/* <button onClick={handleAddCategory} >Agregar</button> */}
      <hr />
      <ol>
        {categories.map(category => (
          <GifGrid
            key={category}
            category={category} />
        ))}
      </ol>
    </>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}

export default GifExpertApp;