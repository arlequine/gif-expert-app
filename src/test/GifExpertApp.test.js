import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Pruebas del componente <GifExpertApp />', () => {
  test('Debe mostrarse correctamente el componente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar una listq de categorias', () => {
    const categories = ['One Piece', 'Dragon ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  });

})
