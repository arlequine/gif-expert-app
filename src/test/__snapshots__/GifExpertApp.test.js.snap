// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> Debe mostrarse correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> debe mostrar una listq de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
